### PR TITLE
Make several CLI commands testable

### PR DESCRIPTION
**Pull request by pcalcado**
As I work on the diagnosis CLI experience, I had to go through our current CLI commands and understand them better. I've written tests and refactored things a bit to document this understanding.

### CODE REVIEW

File: cli/cmd/version.go [hunk 1]
```diff
@@ -4,24 +4,37 @@ import (
...
 
+const DefaultVersionString = "unavailable"
+
 var versionCmd = &cobra.Command{
 	Use:   "version",
 	Short: "Print the client and server version information",
 	Long:  "Print the client and server version information.",
 	Args:  cobra.NoArgs,
 	Run: exitSilentlyOnError(func(cmd *cobra.Command, args []string) error {
-		fmt.Println("Client version: " + controller.Version)
```
[the remark this answers is not in the document]
Comment (pcalcado) [reply cli/cmd/version.go]: @siggy 
 It still prints the version:
```
$ conduit version
Client version: v0.1.0
Server version: v0.1.0
$ bin/go-run cli version
Client version: v0.1.1
Server version: v0.1.0
$ minikube stop > /dev/null
$ bin/go-run cli version
github.com/runconduit/conduit/cli
Client version: v0.1.1
Server version: unavailable
```

The difference, I believe, is that `err` here means "could not instantiate kubectl", not "could not get version from server". The logic that handles errors from the API was moved here: https://github.com/runconduit/conduit/pull/86/files/c49c7e738bef442dbaba164f812c85a25b8d3144#diff-b784f3ae92cb061761114526d43f2e33R58 and is tested here: https://github.com/runconduit/conduit/pull/86/files/c49c7e738bef442dbaba164f812c85a25b8d3144#diff-edf10645fffdcf276967004f4a31ac4fR29

File: cli/cmd/root.go [hunk 2]
```diff
@@ -34,12 +33,7 @@ func addControlPlaneNetworkingArgs(cmd *cobra.Command) {
 	cmd.PersistentFlags().StringVar(&apiAddr, "api-addr", "", "Override kubeconfig and communicate directly with the control plane at host:port (mostly for testing)")
 }
 
-func newApiClient() (pb.ApiClient, error) {
-	kubeApi, err := k8s.MakeK8sAPi(shell.MakeUnixShell(), kubeconfigPath, apiAddr)
-	if err != nil {
-		return nil, err
-	}
-
+func newApiClient(kubeApi k8s.KubernetesApi) (pb.ApiClient, error) {
```
[the remark this answers is not in the document]
Comment (pcalcado) [reply cli/cmd/root.go]: I was trying to be consistent with the way we create other objects. With this PR we have every command instantiating the objects it needs and wiring them together. Arguably this function should not be in the root command but rather follow the pattern established elsewhere and be something like `conduitclient.MakeApiClient(kubeApi k8s.KubernetesApi) (pb.ApiClient, error)`, but I didn't want to propose something like this before there was an actual need for it.

File: cli/cmd/version.go (hunk 1 above)
Comment (siggy) [inline cli/cmd/version.go]: I'm running this from a Docker container, perhaps that's why I see no output?

If either `k8s.MakeK8sAPi()` or `newApiClient()` fail, it prints nothing. I think we should print the client version no matter what, and then perhaps let the user know what went wrong.

**Comment by pcalcado** [comment]
@siggy I can't reproduce the UX issue your reported on https://github.com/runconduit/conduit/pull/86#pullrequestreview-85595353 , but I have the impression that this commit solved the problem: 64e30a50b8877b786cdef9a37f3359fd23d893b6 could you test?

Also, I have no idea why this says the DCO isn't there?

File: cli/cmd/root.go [hunk 3]
```diff
@@ -57,19 +54,18 @@ func newApiClient() (pb.ApiClient, error) {
...
+// that it is a usage error and automatically prints the usage message. `runEButSuppressUsageMessageOnError` has the same signature as `RunE`,
+// but will not display any usage information. It will print the error message to STDERR.
 //
 // TODO: This is used by the `version` command now; it should be used by other commands too.
-func exitSilentlyOnError(f func(cmd *cobra.Command, args []string) error) func(cmd *cobra.Command, args []string) {
+func runEButSuppressUsageMessageOnError(f func(cmd *cobra.Command, args []string) error) func(cmd *cobra.Command, args []string) {
 	return func(cmd *cobra.Command, args []string) {
 		if err := f(cmd, args); err != nil {
-			os.Exit(2) // Reserve 1 for usage errors.
+			fmt.Fprintln(os.Stderr, err)
```
Comment (siggy) [inline cli/cmd/root.go]: this causes errors to be printed twice:
```bash
$ conduit version --kubeconfig foo
error instantiating Kubernetes API client: stat foo: no such file or directory
Client version: v0.1.1
Server version: unavailable
error instantiating Kubernetes API client: stat foo: no such file or directory
```

File: cli/cmd/get_test.go [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+		for i, actualName := range actualPodNames {
+			expectedName := expectedPodNames[i]
+			if expectedName != actualName {
+				t.Fatalf("Expected %dth element on %v to be [%s], but was [%s]", i, actualPodNames, expectedName, actualName)
+			}
+		}
+	})
+
+	t.Run("Returns empty list if no [ods found", func(t *testing.T) {
```
Comment (siggy) [inline cli/cmd/get_test.go]: `[ods`

**Comment by pcalcado** [comment]
@siggy awesome, thanks! I will remove the last commit and later start a conversation around how we do output in the CLI so that we have a standard way to print

**Pull request merged**